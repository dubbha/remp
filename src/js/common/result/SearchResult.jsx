import React from 'react';
import PropTypes from 'prop-types';
import SearchSortByLink from './SearchSortByLink';
import './style.sass';

const SearchResult = ({
  results,
  sortBy,
  onSortByChange,
  sortByParams,
}) => (
  results.length > 0 && (
    <div className="searchResult">
      <span className="searchResult__line">{`${results.length} movies found`}</span>
      <span className="searchResult__line">
        Sort by
        {
          sortByParams.map(item => (
            <SearchSortByLink
              key={item}
              text={item}
              active={sortBy === item}
              onClickWithText={onSortByChange}
            />
          ))
        }
      </span>
    </div>
  )
);

SearchResult.defaultProps = {
  results: [],
};

SearchResult.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape({
    poster: PropTypes.string,
    title: PropTypes.string,
    rating: PropTypes.string,
    category: PropTypes.string,
    release_year: PropTypes.string,
    runtime: PropTypes.string,
    summary: PropTypes.string,
    director: PropTypes.string,
    show_cast: PropTypes.string,
    show_id: PropTypes.number,
  })),
  sortBy: PropTypes.string.isRequired,
  onSortByChange: PropTypes.func.isRequired,
  sortByParams: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default SearchResult;
