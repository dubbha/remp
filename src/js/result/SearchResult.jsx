import React from 'react';
import PropTypes from 'prop-types';
import './style.sass';

const SearchResult = ({ results }) => (
  results.length > 0 && (
    <div className="searchResult">
      <span className="searchResult__line">{`${results.length} movies found`}</span>
      <span className="searchResult__line">
        Sort by
        <a className="searchResult__link">release date</a>
        <a className="searchResult__link searchResult__link_active">rating</a>
      </span>
    </div>
  )
);

SearchResult.defaultProps = {
  results: [],
};

SearchResult.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape({
    poster: PropTypes.string,
    title: PropTypes.string,
    rating: PropTypes.string,
    category: PropTypes.string,
    release_year: PropTypes.string,
    runtime: PropTypes.string,
    summary: PropTypes.string,
    director: PropTypes.string,
    show_cast: PropTypes.string,
    show_id: PropTypes.number,
  })),
};

export default SearchResult;
