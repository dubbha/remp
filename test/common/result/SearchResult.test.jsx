import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchResult from 'common/result/SearchResult';

describe('SearchResult', () => {
  let props;

  beforeEach(() => {
    props = {
      results: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }],
      sortBy: 'release date',
      sortByParams: ['release date', 'rating'],
      onSortByChange: jest.fn(),
    };
  });

  it('should render successfully', () => {
    const wrapper = shallow(<SearchResult {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should switch the active link according to the selected sort by parameter', () => {
    props.sortBy = 'rating';
    const wrapper = shallow(<SearchResult {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render emtpy on empty results list', () => {
    props.results = [];
    const wrapper = shallow(<SearchResult {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
