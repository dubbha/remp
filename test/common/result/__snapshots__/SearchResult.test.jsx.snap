// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should render emtpy on empty results list 1`] = `""`;

exports[`SearchResult should render successfully 1`] = `
<div
  className="searchResult"
>
  <span
    className="searchResult__line"
  >
    5 movies found
  </span>
  <span
    className="searchResult__line"
  >
    Sort by
    <SearchSortByLink
      active={true}
      key="release date"
      onClickWithText={[Function]}
      text="release date"
    />
    <SearchSortByLink
      active={false}
      key="rating"
      onClickWithText={[Function]}
      text="rating"
    />
  </span>
</div>
`;

exports[`SearchResult should switch the active link according to the selected sort by parameter 1`] = `
<div
  className="searchResult"
>
  <span
    className="searchResult__line"
  >
    5 movies found
  </span>
  <span
    className="searchResult__line"
  >
    Sort by
    <SearchSortByLink
      active={false}
      key="release date"
      onClickWithText={[Function]}
      text="release date"
    />
    <SearchSortByLink
      active={true}
      key="rating"
      onClickWithText={[Function]}
      text="rating"
    />
  </span>
</div>
`;
